src/sentry/api/endpoints/group_notes.py

```diff
@@ -31,11 +31,28 @@ def get(self, request, group):
         )
 
     def post(self, request, group):
-        serializer = NoteSerializer(data=request.DATA)
+        serializer = NoteSerializer(data=request.DATA, context={'group': group})
+
         if not serializer.is_valid():
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
         data = dict(serializer.object)
+        mentions = data.pop('mentions', [])
+
+        GroupSubscription.objects.subscribe(
+            group=group,
+            user=request.user,
+            reason=GroupSubscriptionReason.comment,
+        )
+
+        if mentions:
+            users = User.objects.filter(id__in=mentions)
+            for user in users:
+                GroupSubscription.objects.subscribe(
+                    group=group,
+                    user=user,
+                    reason=GroupSubscriptionReason.mentioned,
+                )
 
         if Activity.objects.filter(
```

Comment:
i think you want to do this check before subscribing the user or the mentioned user. so i think you should rearrange this to be something like:
```python
mentions = data.pop...

if Activity.objects.filter(...

GroupSubscription.objects.subscribe(...
```